src/engine/source/cmds/src/test.cpp

```diff
@@ -222,93 +250,107 @@ void run(const Options& options)
-                    if (builder::Asset::Type::RULE == env.assets()[asset]->m_type)
+                    // Rule history
+                    std::cerr << std::endl << "RULES:" << std::endl;
+                    indent = "  ";
+                    for (auto& [asset, condition] : history)
                     {
-                        std::cerr << fmt::format("{}{}  ->  {}", indent, asset, condition) << std::endl;
-                        if (traceBuffer.find(asset) != traceBuffer.end())
+                        if (builder::Asset::Type::RULE == env.assets()[asset]->m_type)
                         {
-                            std::string line;
-                            while (std::getline(traceBuffer[asset], line))
+                            std::cerr << fmt::format("{}{}  ->  {}", indent, asset, condition) << std::endl;
+                            if (traceBuffer.find(asset) != traceBuffer.end())
                             {
-                                std::cerr << indent << indent << line << std::endl;
+                                std::string line;
+                                while (std::getline(traceBuffer[asset], line))
+                                {
+                                    std::cerr << indent << indent << line << std::endl;
+                                }
                             }
-                        }
 
-                        // Clear trace buffer
-                        traceBuffer[asset].str("");
-                        traceBuffer[asset].clear();
-                        if ("success" == condition)
-                        {
-                            indent += indent;
+                            // Clear trace buffer
+                            traceBuffer[asset].str("");
+                            traceBuffer[asset].clear();
+                            if ("success" == condition)
+                            {
+                                indent += indent;
+                            }
                         }
                     }
                 }
-            }
 
-            // Output
-            std::cerr << std::endl << std::endl << "OUTPUT:" << std::endl << std::endl;
-            std::cerr << output.str() << std::endl;
+                // Output
+                std::cerr << std::endl << std::endl << "OUTPUT:" << std::endl << std::endl;
+                std::cerr << output.str() << std::endl;
+            }
+            catch (const std::exception& e)
+            {
+                LOG_ERROR("Engine 'test' command: An error occurred while parsing a message: {}.", e.what());
+            }
         }
-        catch (const std::exception& e)
+        else
         {
-            LOG_ERROR("Engine 'test' command: An error occurred while parsing a message: {}.", e.what());
+            std::cout << std::endl << std::endl << "Enter a log in single line (Crtl+C to exit):" << std::endl << std::endl;
         }
+
     }
 
+    clear_icanon(false);
```

Comment:
For consistency, this section should also include the following condition:

`if (isatty(fileno(stdin)))`